src/internal/git/utils.go

```diff
@@ -43,33 +48,33 @@ func MutateGitUrlsInText(host string, text string, gitUser string) string {
 }
 
 func transformURLtoRepoName(url string) (string, error) {
-	// For further explanation: https://regex101.com/library/UfILls and https://regex101.com/rary/UfILls
-	findRegex := regexp.MustCompile(`\/([\w\-]+)(.git)?(@([\w\-\.]+))?$`)
-	substrings := findRegex.FindStringSubmatch(url)
-	if len(substrings) == 0 {
-		// the first element in the return substrings is
+	matches := gitURLRegex.FindStringSubmatch(url)
+	if len(matches) != 6 {
+		// Unable to find a substring match for the regex
 		return "", fmt.Errorf("unable to get extract the repoName from the url %s", url)
 	}
 
-	// NOTE: The first element in the returned substrings is the combination of all the rest of the substrings....
-	//       So just skip the first element so we can get a hash without the version tag
-	repoName := substrings[1]
+	repoName := matches[gitURLRegex.SubexpIndex("repo")]
+	// NOTE: We remove the .git so that https://zarf.dev/repo.git and https://zarf.dev/repo resolve to the same repo
+	// (as they would in real life)
+	sanitizedURL := matches[gitURLRegex.SubexpIndex("baseURL")] + "/" + repoName + matches[gitURLRegex.SubexpIndex("atRef")]
```

Comment:
@jeff-mccoy here is what changed to make the repo SHAs different.  We remove the .git so that zarf.git and zarf are the same.